xwiki-platform-core/xwiki-platform-comment/xwiki-platform-comment-api/src/main/java/org/xwiki/platform/comment/CommentManager.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.comment;
+
+import java.util.List;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.model.reference.EntityReference;
+
+@Role
+public interface CommentManager
+{
+    List<Comment> getComments(EntityReference documentReference, boolean ascendindOrder) throws CommentException;
```

Comment:
We might want something a bit more advanced in terms of input to cover stuff like pagination in the future.

Reply:
Right I guess I'd need offset and limit in the parameter and probably something for what's returned, or do you think about something else?

Reply:
> or do you think about something else?

Maybe one day we'll need stuff like filtering, but it can be added later. We might want to pass a `CommentQuery` to have a more future-proof `getComments`.